nevergrad/optimization/optimizerlib.py

```diff
@@ -1534,6 +1540,220 @@ def __init__(
 BO = ParametrizedBO().set_name("BO", register=True)
 
 
+class _HyperOpt(base.Optimizer):
+    # pylint: disable=too-many-instance-attributes
+    def __init__(self, parametrization: IntOrParameter,
+                 budget: tp.Optional[int] = None,
+                 num_workers: int = 1,
+                 *,
+                 prior_weight: float = 1.0,
+                 n_startup_jobs: int = 20,
+                 n_EI_candidates: int = 24,
+                 gamma: float = 0.25,
+                 verbose: bool = False) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers)
+        try:
+            # try to convert parametrization to hyperopt search space
+            if not isinstance(parametrization, p.Instrumentation): raise NotImplementedError
```

Comment:
please check `self.parametrization` instead for consistency. The call to `__init__` converts integer inputs to `Array`, so you should not need this special case as it should be handled the same way as `Array` (which if I understand well is not supported either if not bounded)